x/swap/types/pair.go

```diff
@@ -0,0 +1,87 @@
+package types
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Pair defines a tradable token pair
+type Pair struct {
+	TokenA string `json:"token_a" yaml:"token_a"`
+	TokenB string `json:"token_b" yaml:"token_b"`
+}
+
+// NewPair returns a new Pair object
+func NewPair(tokenA, tokenB string) Pair {
+	return Pair{
+		TokenA: tokenA,
+		TokenB: tokenB,
+	}
+}
+
+// Validate validates pair attributes and returns an error if invalid
+func (p Pair) Validate() error {
+	err := sdk.ValidateDenom(p.TokenA)
+	if err != nil {
+		return err
+	}
+
+	err = sdk.ValidateDenom(p.TokenB)
+	if err != nil {
+		return err
+	}
+
+	if p.TokenA == p.TokenB {
+		return fmt.Errorf(
+			"pair cannot have two tokens of the same type, received '%s' and '%s'",
+			p.TokenA, p.TokenB,
+		)
+	}
+
+	return nil
+}
+
+// Name returns a unique name for a pair in alphabetical order
+func (p Pair) Name() string {
+	if p.TokenA < p.TokenB {
```

Comment:
@karzak @denalimarsh @rhuairahrighairidh Should we move this to Validate?  It seems like we could that `TokenA <TokenB` at param creation, and clients can always assume the name will be `:token_a:/:token_b:`.